crates/sc-consensus-subspace/src/archiver.rs

```diff
@@ -697,14 +708,6 @@ where
             }
 
             if !new_segment_headers.is_empty() {
-                if let Err(error) = segment_headers_store.add_segment_headers(&new_segment_headers)
```

Comment:
Is the moving up of this for a race condition where something happens in the time window after notification is sent, and before it is saved?

Reply:
No race condition, it is fully deterministic. Notification we send above will wait for archived segment to be acknowledged. Farmer is doing some things before acknowledging, one of which might try to retrieve it, but it was only stored after all notifications were acknowledged, breaking the logic on the farmer side in some edge cases.